### PR TITLE
Move most of `rebase_commit_with_options()` onto new `CommitRewriter` type

### PR DESCRIPTION
**Pull request by martinvonz**
# Checklist

If applicable:
- [ ] I have updated `CHANGELOG.md`
- [ ] I have updated the documentation (README.md, docs/, demos/)
- [ ] I have updated the config schema (cli/src/config-schema.json)
- [ ] I have added tests to cover my changes

### CODE REVIEW

**Comment by martinvonz** [comment]
@bnjmnt4n: This PR implements gets very close to the `transform_descendants()` function I've been talking about, so maybe you'll want to take a look

File: lib/src/rewrite.rs [hunk 1]
```diff
@@ -111,6 +112,56 @@ pub fn rebase_commit(
...
+        self.mut_repo
+    }
+
+    /// The commit we're rewriting.
+    pub fn old_commit(&self) -> &Commit {
+        &self.old_commit
+    }
+
+    /// Get the old commit's intended new parents.
+    pub fn new_parents(&self) -> &Vec<CommitId> {
```
Comment (yuja) [inline lib/src/rewrite.rs:139]: nit: can return slice `&[CommitId]`

File: lib/src/rewrite.rs [hunk 2]
```diff
@@ -145,6 +145,12 @@ impl<'repo> CommitRewriter<'repo> {
         self.new_parents = new_parents;
     }
 
+    /// Checks if the intended new parents are different from the old commit's
+    /// parents.
+    pub fn parents_changed(&self) -> bool {
+        self.new_parents.as_slice() != self.old_commit.parent_ids()
```
Comment (yuja) [inline lib/src/rewrite.rs:151]: nit: `.as_slice()` can be omitted?

**Review by yuja — APPROVED** [review]

File: lib/src/rewrite.rs (hunk 2 above)
Comment (martinvonz) [reply lib/src/rewrite.rs]: Done

**Pull request merged**